src/Maker/MakeResetPassword.php

```diff
@@ -194,23 +203,94 @@ public function generate(InputInterface $input, ConsoleStyle $io, Generator $gen
+
+        $pathRequestEntity = $entityClassGenerator->generateEntityClass($requestClassNameDetails, false);
+
+        $generator->writeChanges();
+
+        $manipulator = new ClassSourceManipulator(
+            $this->fileManager->getFileContents($pathRequestEntity)
         );
 
-        $generator->generateClass(
-            $repositoryClassNameDetails->getFullName(),
-            'resetPassword/ResetPasswordRequestRepository.tpl.php',
-            [
-                'request_class_full_name' => $requestClassNameDetails->getFullName(),
-                'request_class_name' => $requestClassNameDetails->getShortName(),
-            ]
+        $manipulator->addUseStatementIfNecessary(ResetPasswordRequestInterface::class);
+
+        $manipulator->addUseStatementIfNecessary(ResetPasswordRequestTrait::class);
+
+        $manipulator->addInterface('ResetPasswordRequestInterface');
+
+        $manipulator->clearClassNodeStmts();
+
+        $manipulator->addTrait('ResetPasswordRequestTrait');
+
+        $manipulator->addConstructor([
+                ['user', null, 'object'],
+                ['expiresAt', null, '\DateTimeInterface'],
+                ['selector', null, 'string'],
+                ['hashedToken', null, 'string'],
+            ], <<<'CODE'
+<?php
+$this->user = $user;
+$this->initialize($expiresAt, $selector, $hashedToken);
+CODE
+        );
+        $manipulator->addProperty('id', [
+            '@ORM\Id()',
+            '@ORM\GeneratedValue()',
+            '@ORM\Column(type="integer")',
+        ]);
+
+        $manipulator->addProperty('user', [
+            '@ORM\ManyToOne(targetEntity="App\Entity\User")',
+        ]);
+
+        $manipulator->addGetter('user', 'object', false);
+
+        $this->fileManager->dumpFile(
+            $pathRequestEntity,
+            $manipulator->getSourceCode()
+        );
+
+        $pathRequestRepository = $this->fileManager->getRelativePathForFutureClass(
+            $repositoryClassNameDetails->getFullName()
+        );
+
+        $manipulator = new ClassSourceManipulator(
+            $this->fileManager->getFileContents($pathRequestRepository)
+        );
+
+        $manipulator->clearClassNodeStmts();
+
+        $manipulator->addUseStatementIfNecessary(ResetPasswordRequestInterface::class);
+
+        $manipulator->addUseStatementIfNecessary(ResetPasswordRequestRepositoryTrait::class);
+
+        $manipulator->addUseStatementIfNecessary(ResetPasswordRequestRepositoryInterface::class);
+
+        $manipulator->addInterface('ResetPasswordRequestRepositoryInterface');
+
+        $manipulator->addTrait('ResetPasswordRequestRepositoryTrait');
+
+        $manipulator->addConstructor([
+                ['registry', null, 'ManagerRegistry'],
+            ], <<<'CODE'
+<?php
+parent::__construct($registry, ResetPasswordRequest::class);
+CODE
+        );
+        $methodBuilder = $manipulator->createMethodBuilder('createResetPasswordRequest', 'ResetPasswordRequestInterface', false);
```

Comment:
I wrote this method originally (I see), but now that I look at it, I think the 2nd argument to this method should be the full class/interface name of the return type (for cases where it's a class/interface) - so `ResetPasswordRequestInterface::class`. THEN, the `createMethodBuilder()` internally should check to see if the passed returnType is a class or interface (check both `class_exists()` and `interface_exists()`) and add the `use` statement automatically IF it is a class or interface.

Reply:
Done